src/cgroups/v1/devices.rs

```diff
@@ -98,24 +98,123 @@ impl Devices {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::cgroups::test::{create_temp_dir, set_fixture};
+    use oci_spec::{LinuxDeviceCgroup, LinuxDeviceType};
 
     #[test]
-    fn test_linux_device_cgroup_to_string() {
-        let ldc = LinuxDeviceCgroup {
-            allow: true,
-            typ: LinuxDeviceType::A,
-            major: None,
-            minor: None,
-            access: "rwm".into(),
-        };
-        assert_eq!(ldc.to_string(), "a *:* rwm");
-        let ldc = LinuxDeviceCgroup {
-            allow: true,
-            typ: LinuxDeviceType::A,
-            major: Some(1),
-            minor: Some(9),
-            access: "rwm".into(),
-        };
-        assert_eq!(ldc.to_string(), "a 1:9 rwm");
+    fn test_set_default_devices() {
+        let tmp =
+            create_temp_dir("test_set_default_devices").expect("create temp directory for test");
+
+        Devices::default_allow_devices().iter().for_each(|d| {
+            // NOTE: We reset the fixtures every iteration because files aren't appended
+            // so what happens in the tests is you get strange overwrites which can contain
+            // remaining bytes from the last iteration. Resetting the files more appropriately
+            // mocks the behavior of cgroup files.
+            set_fixture(&tmp, "devices.allow", "").expect("create allowed devices list");
+            set_fixture(&tmp, "devices.deny", "").expect("create denied devices list");
+
+            Devices::apply_device(&d, &tmp).expect("Apply default device");
+            println!("Device: {}", d.to_string());
+            if d.allow {
+                let allowed_content =
+                    std::fs::read_to_string(tmp.join("devices.allow")).expect("read to string");
```

Comment:
I  felt it easier to read the `read_to_string` import in the test scope.

Reply:
I started thinking the same thing. At first it was only used a few times, but now it's getting pretty verbose.

Reply:
I am now importing `read_to_string` in to all of the tests I had touched with this PR.